memlru/memlru.go

```diff
@@ -116,6 +117,24 @@ func (m *MemLRU[V]) BatchSetEx(ctx context.Context, keys []string, values []V, t
 	return nil
 }
 
+func (c *MemLRU[V]) GetEx(ctx context.Context, key string) (out V, ttl time.Duration, exists bool, err error) {
+	out, exists, err = c.Get(ctx, key)
+	if err != nil {
+		return out, 0, false, fmt.Errorf("get %w", err)
+	}
+
+	if !exists {
+		return out, 0, false, nil
+	}
+
+	item, ok := c.expirationQueue.GetItem(key)
+	if !ok {
+		return out, time.Duration(math.MaxInt64), true, nil
```

Comment:
Max duration was my idea.. but I see how it might be confusing.

However, `nil` pointer might be annoying from the client perspective too.

Reply:
When I think about this from the user perspective.. I think I need expiry mostly so I can set `Cache-Control: max-age=604800` header

![Screenshot 2024-09-19 at 2 33 25 PM](https://github.com/user-attachments/assets/f78c87e1-b45c-437a-9f98-f1b722852849)

Reply:
I imagined users would be writing something like this:

```go
v, ttl, ok, err := cache.GetEx(ctx, "some-key")
if err != nil {
      return err
}

if ttl > 0 {
   w.Header().Set("max-age", fmt.Sprintf("%v, ttl.Seconds())
}

// w.Write(w.Bytes())
```

What would you propose in case of "no expiry" in cachestore? How would you handle it? I think you'd want to cache it (`max-age`) forever.. right?

But yeah.. you might be right that explicit return value would be better

Reply:
I get your use case, but when i get the `ttl` larger then 0 i expect, that the expiration on that object exists, and that is confusing for me.

Use case:
I can imagine, that somebody would schedule the job for updating the cache based on TTL, so he counts the time, and do the updating some moment before (because the updating job can take a long time and the dev want to keep the functions fast). This way the dev would do the updating even tho it is not needed.

Yes, it is edge case, but it can be a case, i am just thinking.

Devs can do this:
```go
if ttl != nil {
  w.Header().Set("max-age", fmt.Sprintf("%v, ttl.Seconds())
} else {
  w.Header().Set("max-age", fmt.Sprintf("%v, myOwnMaxIntForCache)
}
```

Reply:
Or, to prevent the `nil` checks, we could go for negative duration in case of no expiry:
```go
if ttl >= 0 {
  w.Header().Set("max-age", fmt.Sprintf("%v, ttl.Seconds())
} else {
  w.Header().Set("max-age", fmt.Sprintf("%v, myOwnMaxCache)
}
```